### ISSUE TITLE
Mock method backward compatibility 3.3.1 and 3.4.1 versions

### ISSUE THREAD

**Issue by berehovskyi** [ApexTestKit/ApexTestKit#31]
Using `3.3.1` version the next code gives the result:
```java
Id userId = '005000000000003AAA';
Id managerId = '005000000000004AAA';
List<User> users = (List<User>) ATK.prepare(User.SObjectType, 1)
        .profile('System Administrator')
        .field(User.Id).repeat(userId)
        .field(User.LastName).repeat('userLastName')
        .field(User.Name).repeat('userName')
        .field(User.ManagerId).repeat(managerId)
        .mock()
        .get(User.SObjectType);
System.debug(LoggingLevel.INFO, 'users ' + JSON.serializePretty(users));
/*
{
  "attributes" : {
    "type" : "User",
    "url" : "/services/data/v51.0/sobjects/User/005000000000003AAA"
  },
  "ProfileId" : "00e1j000000htRQAAY",
  "Id" : "005000000000003AAA",
  "LastName" : "userLastName",
  "ManagerId" : "005000000000004AAA",
  "Name" : "userName"
}
*/
```
Using `3.4.1` version the next code gives the result:
```java
Id userId = '005000000000003AAA';
Id managerId = '005000000000004AAA';
List<User> users = (List<User>) ATK.prepare(User.SObjectType, 1)
        .profile('System Administrator')
        .field(User.Id).repeat(userId)
        .field(User.LastName).repeat('userLastName')
        .field(User.Name).repeat('userName')
        .field(User.ManagerId).repeat(managerId)
        .mock()
        .get(User.SObjectType);
System.debug(LoggingLevel.INFO, 'users ' + JSON.serializePretty(users));
/*
{
  "attributes" : {
    "type" : "User",
    "url" : "/services/data/v51.0/sobjects/User/005000zzzz00001AAA"
  },
  "ProfileId" : "00e1j000000htRQAAY",
  "Id" : "005000zzzz00001AAA",
  "LastName" : "userLastName",
  "ManagerId" : "005000000000004AAA",
  "Name" : "userName"
}
*/
```
Somehow in `3.4.1` the `User.Id` field is being overridden by the newly generated `Id` value regardless we use `.field(User.Id).repeat(userId)` method invocation or not. Although `ManagerId` lookup wasn't overridden.
Is it expected behavior and what should we do in case if we want to set the `Id` from the outside?
I've raised the issue at least because the `3.4.1` version breaks the backward compatibility with `3.3.1` for such a specific scenario.

**Comment by inksword** [comment]
@berehovskyi Thanks for raising this issue. Will deal with it as soon as possible.

**Comment by inksword** [comment]
@berehovskyi The issue has been fixed with v3.4.2. I will reopen this ticket if there is other side effects.

You can also try to use `ATK.fakeId(User.SObjectType)` to generate a fake user id. Please also checkout the Fake Id section.

I have also added a unit test to cover the case:

```java
    @IsTest
    static void test_Mock_PredefinedId() {
        Id accountId0 = ATK.fakeId(Account.SObjectType);
        Id accountId1 = ATK.fakeId(Account.SObjectType);
        Id contactId0 = ATK.fakeId(Contact.SObjectType);
        Id contactId1 = ATK.fakeId(Contact.SObjectType);

        ATK.SaveResult result = ATK.prepare(Account.SObjectType, 2)
            .field(Account.Id).repeat(accountId0, (String)accountId1)
            .withChildren(Contact.SObjectType, Contact.AccountId, 2)
                .field(Contact.Id).repeat(contactId0, (String)contactId1)
            .mock();

        System.assertEquals(accountId0, result.get(Account.SObjectType)[0].Id);
        System.assertEquals(accountId1, result.get(Account.SObjectType)[1].Id);
        System.assertEquals(contactId0, result.get(Contact.SObjectType)[0].Id);
        System.assertEquals(contactId1, result.get(Contact.SObjectType)[1].Id);
        System.assertEquals(accountId0, ((Contact)result.get(Contact.SObjectType)[0]).AccountId);
        System.assertEquals(accountId1, ((Contact)result.get(Contact.SObjectType)[1]).AccountId);
    }
```
